File: src/game/components/player.lua
local constants = require 'game.components.constants'

local Player = {
  _meta = constants.Player
}

function Player.new(id, alias, money, lives, documents, checkpoint)
  return {
    _meta = Player._meta,
    id = id,
    alias = alias or '',
    money = money or 0,
    lives = lives or 100,
    documents = documents or 0,
    checkpoint = checkpoint or 1
  }
end

return Player

File: client2/src/game/components/player.ts
import { ComponentFlag } from '../flags'

export class Player {
  static _id = 'Player'
  _id = Player._id

  static _flag = ComponentFlag.Player
  _flag = ComponentFlag.Player

  alias: string
  money: number
  lives: number
  documents: number
  checkpoint: number

  constructor(
    alias: string = '',
    money: number = 0,
    lives: number = 0,
    documents: number = 0,
    checkpoint: number = 0
  ) {
    this.alias = alias
    this.money = money
    this.lives = lives
    this.documents = documents
    this.checkpoint = checkpoint
  }
}
